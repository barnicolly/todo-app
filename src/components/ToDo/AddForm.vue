<template>
  <form @submit.prevent="onSubmit">
    <input
      type="text"
      autocomplete="off"
      v-model.trim="title"
      placeholder="Новая задача" />
    <button class="btn btn--add" type="button" @click="onSubmit">
      <font-awesome-icon icon="plus" />
    </button>
  </form>
</template>

<script setup lang="ts">
import { useTodoStore } from '@/store/todo';
import { ModelRef } from 'vue';
import { useAlertStore } from '@/store/alert';

const title: ModelRef<string, string> = defineModel({
  default: '',
});
const store = useTodoStore();
const alertStore = useAlertStore();

function onSubmit() {
  if (title.value === undefined || title.value === '') {
    alertStore.showAlert('Введите текст новой задачи');
    return;
  }
  store.addToDo(title.value);
  alertStore.removeAlert();
  title.value = '';
}
</script>
