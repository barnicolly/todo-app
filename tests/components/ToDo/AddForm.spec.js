import { shallowMount } from '@vue/test-utils';
import { Store } from 'vuex';
import AddForm from '../../../src/components/ToDo/AddForm.vue';

// todo-misha переделать селекторы как будут классы;
describe('AddForm.vue', () => {
  let wrapper;
  const defaultTitle = '';
  const changedTitle = 'changed-title';
  const addToDoMock = jest.fn();
  const $store = new Store({
    mutations: {
      addToDo: addToDoMock,
    },
  });

  beforeEach(() => {
    wrapper = shallowMount(AddForm, {
      global: {
        mocks: {
          $store,
        },
      },
    });
  });

  it('should correct initialize', () => {
    expect(wrapper.vm.title).toBe(defaultTitle);
  });

  it('change title', async () => {
    const input = wrapper.find('input');

    input.element.value = changedTitle;
    await input.trigger('change');

    expect(wrapper.vm.title).toBe(changedTitle);
  });

  it('click add btn', async () => {
    const input = wrapper.find('input');
    input.element.value = changedTitle;
    await input.trigger('change');
    expect(wrapper.vm.title).toBe(changedTitle);

    wrapper.find('button[type="submit"]').trigger('click');
    expect(addToDoMock.mock.calls.length).toBe(1);
    expect(addToDoMock.mock.calls[0][1]).toBe(changedTitle);
    expect(wrapper.vm.title).toBe(defaultTitle);
  });
});

describe('AddForm.vue Snapshot', () => {
  it('has same HTML structure', () => {
    const wrapper = shallowMount(AddForm);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
