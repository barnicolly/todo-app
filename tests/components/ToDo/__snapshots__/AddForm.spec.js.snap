// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddForm.vue Snapshot has same HTML structure 1`] = `
<form><input type="text" autocomplete="off" placeholder="Новая задача"><button class="btn btn--add" type="button">
    <font-awesome-icon-stub icon="plus" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false"></font-awesome-icon-stub>
  </button></form>
`;
